build/lib/src/generate/run_builder.dart

```diff
@@ -62,3 +93,26 @@ Future<void> runBuilder(Builder builder, Iterable<AssetId> inputs,
     await resources.beforeExit();
   }
 }
+
+extension on Package {
+  Package transformToAssetUris() {
+    return Package(
+      name,
+      Uri(scheme: 'asset', pathSegments: [name, '']),
+      packageUriRoot: Uri(scheme: 'asset', pathSegments: [name, 'lib', '']),
+      relativeRoot: false,
+      extraData: extraData,
```

Comment:
```suggestion
      languageVersion: languageVersion,
```

Reply:
Ouch, missed the most important piece of information :D I've also expanded the test to ensure we get the language version.

Should we still remove the `extraData` field?

Reply:
Oh sorry I didn't intend to suggest removing that, just a goof up